src/main/java/com/squareup/javapoet/JavaFile.java

```diff
@@ -47,13 +47,17 @@ private JavaFile(Builder builder) {
   }
 
   public void emit(Appendable out) throws IOException {
+    emit(out, "  "); //TODO(pwittchen): this call is made because of toString() and can be improved
```

Comment:
Just hardcode `toString()` to call the emit overload with `"  "`. That's reasonable.